### ISSUE TITLE
Additional return code for GET /<userid>/storage/<collection>?

### ISSUE THREAD

**Issue by ckarlof** [mozilla/picl-server#28]
For GET /<userid>/storage/<collection>, if a client supplies a version greater than the current version number of the collection, does this indicate an error in terms of the client state? Should we return some 4xx error in this case?

**Comment by rfk** [comment]
This sounds like a sensible precaution, there's no good reason why a client should do that.

**Comment by rfk** [comment]
I can't think of a more appropriate response code than "400 Bad Request", but we could add information in the response body to highlight the source of the error.  @ckarlof @zaach what do you think of this kind of detailed error information:

   https://docs.services.mozilla.com/storage/apis-2.0.html#error-response-format

**Comment by ckarlof** [comment]
400 is fine with me.

**Comment by ckarlof** [comment]
Regarding error formatting, I'm fine with your proposal. I think it's important to have a machine readable field that identifies the particular error. Did you intend for the `status` property to do this?
https://bugzilla.mozilla.org/show_bug.cgi?id=792674#c0

Also, check in with the android team. @seanmonstar

**Comment by zaach** [comment]
Hapi lets you define onPostHandler, which can be used to reformat responses before they are sent. That should be useful here.

**Comment by rfk** [comment]
> I think it's important to have a machine readable field that identifies the particular error.
> Did you intend for the status property to do this?

Yes, precisely - as a human-readable but machine-checkable string rather than an opaque error code.

**Comment by ckarlof** [comment]
> Yes, precisely - as a human-readable but machine-checkable string rather than an opaque error code.

Sounds good. Error handling is a little haphazard on the client side right now, so it will be good to tighten it up. I can work on it after this Friday's demo deadline.
